force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -213,9 +205,11 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
 
-        // Subtract the difference in years
-
-        return null; // Replace null with the variable you used to store the result
+        if(d1 == null || d2 == null){
+            return null;
+        }
+        return (Math.max(d1.year(),d2.year())-Math.min(d1.year(), d2.year())); // Subtract the difference in years + return
```

Comment:
This is super interesting and I can your logic. But you can also get the absolute value by using the `abs()` method. That way it doesn't matter which year is larger (more recent).
```suggestion
        return Math.abs(d1.year() - d2.year());
```